subsys/mgmt/osdp/src/osdp.c

```diff
@@ -199,7 +204,24 @@ static int osdp_init(const struct device *arg)
 
 	/* setup OSDP */
 	ctx = osdp_build_ctx(&channel);
-	if (osdp_setup(ctx)) {
+	if (ctx == NULL) {
+		LOG_ERR("Failed to build osdp ctx");
+		k_panic();
+	}
+
+#ifdef CONFIG_OSDP_SC_ENABLED
+	uint8_t key_buf[16];
+
+	if (strcmp(OSDP_KEY_STRING, "NONE") != 0) {
+		if (osdp_key_string_to_byte_arrray(key_buf, OSDP_KEY_STRING)) {
+			LOG_ERR("Failed to parse key buffer");
+			return -1;
+		}
+		key = key_buf;
```

Comment:
If we use the `IS_ENABLED()` like I suggested above, then this would not work any more as we would have a pointer to stack variable that is going away when exiting this `if` block. So the `key_buf` would be needed to move out of the `if` block in that case.